src/postgres-util/src/tunnel.proto

```diff
@@ -0,0 +1,21 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package mz_postgres_util.tunnel;
+
+message ProtoPostgresFlavor {
```

Comment:
I initially had it as an `enum` but it turns out that it is very annoying to use. When you use this enum as the type of a field  in a proto definition it results in a `i32` Rust type for said field. This means that the `from_proto` implementation of the struct containing the enum field needs to first convert the `i32` into a `ProtoPostgresFlavor` and then convert *that* into `PostgresFlavor`. Apart from being annoying to go through two conversion steps this also doesn't work because the message containing the field is in a different crate than `mz_postgres_util` and doesn't have access to the `ProtoPostgresFlavor` struct since those are kept private in the respective modules. The only salvation is to implement `RustType<i32> for PostgresFlavor` instead of `RustType<ProtoPostgresFlavor>` but I didn't want to contort the trait impl.

So in the end I switched to a message so that foreign crates don't need to explicitly import the proto types and just attempt to decode the field through the trait.

Reply:
gotcha, makes sense!